app/services/usps_in_person_proofing/enrollment_helper.rb

```diff
@@ -1,61 +1,94 @@
-      user.confirmed_email_addresses.each do |email_address|
-        UserMailer.in_person_ready_to_verify(
-          user,
-          email_address,
-          first_name: pii['first_name'],
-          enrollment: enrollment,
-        ).deliver_now_or_later
+        send_ready_to_verify_email(user, pii, enrollment)
       end
-    end
 
-    def usps_proofer
-      if IdentityConfig.store.usps_mock_fallback
-        UspsInPersonProofing::Mock::Proofer.new
-      else
-        UspsInPersonProofing::Proofer.new
+      def send_ready_to_verify_email(user, pii, enrollment)
+        user.confirmed_email_addresses.each do |email_address|
+          UserMailer.in_person_ready_to_verify(
+            user,
+            email_address,
+            first_name: pii['first_name'],
+            enrollment: enrollment,
+          ).deliver_now_or_later
+        end
       end
-    end
 
-    def create_usps_enrollment(enrollment, pii)
-      applicant = UspsInPersonProofing::Applicant.new(
-        {
-          unique_id: enrollment.usps_unique_id,
-          first_name: pii['first_name'],
-          last_name: pii['last_name'],
-          address: pii['address1'],
-          # do we need address2?
-          city: pii['city'],
-          state: pii['state'],
-          zip_code: pii['zipcode'],
-          email: 'no-reply@login.gov',
-        },
-      )
-      proofer = usps_proofer
-
-      response = proofer.request_enroll(applicant)
-      response['enrollmentCode']
+      def establishing_in_person_enrollment_for_user(user)
+        cancel_stale_establishing_enrollments_for_user(user)
+        enrollment = user.establishing_in_person_enrollment
+        return enrollment if enrollment.present?
+
+        InPersonEnrollment.create!(user: user)
+      end
+
+      def usps_proofer
+        if IdentityConfig.store.usps_mock_fallback
+          UspsInPersonProofing::Mock::Proofer.new
+        else
+          UspsInPersonProofing::Proofer.new
+        end
+      end
+
+      def create_usps_enrollment(enrollment, pii)
+        applicant = UspsInPersonProofing::Applicant.new(
+          {
+            unique_id: enrollment.usps_unique_id,
+            first_name: pii['first_name'],
+            last_name: pii['last_name'],
+            address: pii['address1'],
+            # do we need address2?
+            city: pii['city'],
+            state: pii['state'],
+            zip_code: pii['zipcode'],
+            email: 'no-reply@login.gov',
+          },
+        )
+        proofer = usps_proofer
+
+        response = proofer.request_enroll(applicant)
+        response['enrollmentCode']
+      end
+
+      private
+
+      def cancel_stale_establishing_enrollments_for_user(user)
+        # cancel any establishing enrollments that have profiles
+        user.
+          in_person_enrollments.
+          where(status: :establishing).
+          where.not(profile: nil).
+          each(&:cancelled!)
+        # and all but the most recent without profiles
```

Comment:
Could you explain the significance of those with/without profiles and why we want to keep anything?